python/pyspark/sql/connect/client/reattach.py

```diff
@@ -14,14 +14,16 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 #
+from multiprocessing import RLock
```

Comment:
nit but the import has to be below around `from multiprocessing.pool import ThreadPool`

Reply:
Done